src/dev/flang/util/Errors.java

```diff
@@ -454,7 +454,7 @@ public static void showAndExit(boolean warningStatistics)
   {
     if (count() > 0)
       {
-        if (count() >= MAX_ERROR_MESSAGES)
+        if (count() >= MAX_ERROR_MESSAGES && MAX_ERROR_MESSAGES != -1)
```

Comment:
Can you check if this code can be moved to before the call of `showAndExit` in `error` to avoid this second `if` check?